kqueen/provisioners/heat.py

```diff
@@ -0,0 +1,180 @@
+                user_domain_name=kwargs.get('user_domain_name', ''),
+                trust_id=kwargs.get('trust_id', ''),
+                domain_id=kwargs.get('domain_id', ''),
+                domain_name=kwargs.get('domain_name', ''),
+                project_id=kwargs.get('project_id', ''),
+                project_name=kwargs.get('project_name', ''),
+                project_domain_id=kwargs.get('project_domain_id', ''),
+                project_domain_name=kwargs.get('project_domain_name', ''),
+                reauthenticate=kwargs.get('reauthenticate', False))
+
+        sess = session.Session(auth=auth)
+        self.heat_cli = heatclient.client.Client('1', session=sess)
+
+    def __import_clusters(self, clusters):
+        for cluster in clusters:
+            self.clusters[cluster.id] = cluster
+
+    def __process_required_files(self, template_file, env_paths):
+        merged_files, merged_env = template_utils.process_multiple_environments_and_files(env_paths)
+        composite_templates, template = template_utils.process_template_path(template_file)
+        merged_files.update(composite_templates)
+        return template, merged_files, merged_env
+
+    def __get_stack(self, stack_id):
+        return self.heat_cli.stacks.get(stack_id)
+
+    def __create_stack(self, stack_name, template, files, environment):
+        return self.heat_cli.stacks.create(stack_name=stack_name, template=template, files=files, environment=environment)
+
+    def __delete_stack(self, stack_id):
+        return self.heat_cli.stacks.delete(stack_id=stack_id)
+
+    def create(self, stack_name, template_file, env_paths=[]):
+        cluster_id = 'cluster-{}-{}'.format(self.provisioner, stack_name)
+
+        if cluster_id not in self.clusters:
+            # Load all required files for template
+            template, merged_files, merged_env = self.__process_required_files(template_file, env_paths)
+
+            try:
+                # Create stack
+                r = self.__create_stack(stack_name, template, merged_files, merged_env)
+            except Exception as e:
+                logger.error('Stack {} failed to create with reason: {}'.format(stack_name, str(e)))
+                return None
+
+            self.clusters[cluster_id] = {
+                'name': cluster_id,
+                'provisioner': self.provisioner,
+                'state': 'CREATE_IN_PROGRESS',
+                'parameters': {
+                    'stack_id': r['stack']['id'],
+                    'outputs': {},
+                },
+                'kubeconfig': ''
+            }
+        else:
+            logger.error('Stack {} already exists. Create aborted'.format(stack_name))
+
+        return self.clusters[cluster_id]
+
+    def get(self, cluster_id, cacheEnabled=True, kubeconfig_output_key='kubeconfig'):
+        if cacheEnabled and cache.get(cluster_id):
+            return cache.get(cluster_id)
+
+        if cluster_id in self.clusters:
```

Comment:
you can negate here, so it will be 
``` python
if clusterd_id not in self.clusters:
  logger.error('Cluster {} not managed by this provisioner'.format(cluster_id))
  return None
```
and continue with less nesting.